sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareAsyncClient.java

```diff
@@ -780,21 +790,59 @@ public ShareFileAsyncClient getFileClient(String filePath) {
     public Mono<Response<ShareFileAsyncClient>> createFileWithResponse(String fileName, long maxSize,
         ShareFileHttpHeaders httpHeaders, FileSmbProperties smbProperties, String filePermission,
         Map<String, String> metadata) {
+        return this.createFileWithResponse(fileName, maxSize, httpHeaders, smbProperties, filePermission, metadata,
+            null);
+    }
+
+    /**
+     * Creates the file in the share with the given name, file max size and associates the passed properties to it.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Create the file "myfile" with length of 1024 bytes, some headers, file smb properties and metadata</p>
+     *
+     * {@codesnippet com.azure.storage.file.share.ShareAsyncClient.createFileWithResponse#String-long-ShareFileHttpHeaders-FileSmbProperties-String-Map-ShareRequestConditions}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-file">Azure Docs</a>.</p>
+     *
+     * @param fileName Name of the file.
+     * @param maxSize The maximum size in bytes for the file, up to 1 TiB.
+     * @param httpHeaders The user settable file http headers.
+     * @param smbProperties The user settable file smb properties.
+     * @param filePermission The file permission of the file.
+     * @param metadata Optional name-value pairs associated with the file as metadata.
+     * @param requestConditions {@link ShareRequestConditions}
+     * @return A response containing a {@link ShareFileAsyncClient} to interact with the created file and the status of
+     * its creation.
+     * @throws ShareStorageException If one of the following cases happen:
+     * <ul>
+     * <li>
+     * If the share or parent directory does not exist.
+     * </li>
+     * <li>
+     * An attempt to create file on a share snapshot will fail with 400 (InvalidQueryParameterValue).
+     * </li>
+     * </ul>
+     */
+    public Mono<Response<ShareFileAsyncClient>> createFileWithResponse(String fileName, long maxSize,
+        ShareFileHttpHeaders httpHeaders, FileSmbProperties smbProperties, String filePermission,
+        Map<String, String> metadata, ShareRequestConditions requestConditions) {
```

Comment:
Is File leasing new as of `2019-07-07`? 

I heard there were discussions about potentially having newer APIs throw if they are supported by older service versions and the client is using said version. Would this need to throw if the client was using service version `2019-02-02`?

Reply:
It is new for 2019-07-07. I haven't been clear about the story for specifying the versions. I heard what you mentioned. I also heard something about just clearing the list every time there's a service change that's not supported by old versions? Where can we go to get clarity on that?

Reply:
We had a conversation about this this morning. I think we are going to come up with a more complete approach to doing service version validation and apply that to all Jul19 features at once, so I will defer the check until that time.